src/docfx/lib/github/UserProfile.cs

```diff
@@ -24,5 +27,29 @@ internal class UserProfile
 
         [JsonProperty("user_emails")]
         public string UserEmails { get; set; }
+
+        public string[] GetUserEmails() => UserEmails != null ? UserEmails.Split(";") : Array.Empty<string>();
+
+        public UserProfile AddEmail(string email)
+        {
+            return new UserProfile
+            {
+                ProfileUrl = ProfileUrl,
+                DisplayName = DisplayName,
+                Name = Name,
+                Id = Id,
+                EmailAddress = EmailAddress,
+                UserEmails = Merge(GetUserEmails(), email),
+            };
+
+            string Merge(string[] existingEmails, string newEmail)
+            {
+                var result = new HashSet<string>(existingEmails)
+                {
+                    newEmail,
+                };
+                return string.Join(';', result);
```

Comment:
>Join [](start = 30, length = 4)

can we avoid these joins and splits when updating cache?

Reply:
Add a TODO to handle this

---
In reply to: [210178519](https://github.com/dotnet/docfx/pull/3176#discussion_r210178519) [](ancestors = 210178519)